app/controllers/api/lessons_controller.rb

```diff
@@ -16,9 +16,26 @@ def show
     end
 
     def create
-      @lesson = Lesson.new params.permit :group_id, :date, :subject_id
-      @lesson.save
-      respond_with @lesson, include: included_params, meta: { timestamp: Time.zone.now }
+      lesson = Lesson.find_by group_id: params[:group_id], subject_id: params[:subject_id], date: params[:date]
```

Comment:
This is unfortunately a race condition, you really want to leverage the database to get this right - e.g: constraint violation.